### PR TITLE
i18n(ja): Add new `docker.mdx` translation

### PR DESCRIPTION
**Pull request by piro-hiroki**
#### Description (required)

Translate `docker.mdx` into Japanese！
file: [recipes/docker.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/recipes/docker.mdx)

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Comment by piro-hiroki** [comment]
`ja/guides/deploy/google-cloud/` の翻訳が最新の変更に追従してなさそう。。。/ The translation of `en/guides/deploy/google-cloud/` doesn't seem to follow the latest changes....

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! Just checking if this PR should still be marked as a draft?

If it's ready for editing, then please change the status so that it is ready for review! We are hoping to merge as many PRs by next week as we can before the docs site code migrates to Starlight. So if this is ready, we would like to be able to review it.  Thanks!

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! I think I fixed the links on this one! Is it ready for a review?

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:19]:
```suggestion
プロジェクトのルートディレクトリに`Dockerfile`というファイルを作成します。このファイルにはサイトを構築するための指示が含まれおり、ニーズに応じて内容は異なります。このガイドではすべての可能なオプションを紹介することはできませんが、SSRと静的モードのスタートポイントを提供します。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:25]:
```suggestion
このDockerfileはNode.jsを使用してサイトを構築し、ポート`4321`でサービスを提供します。そのため、Astroプロジェクトに[Nodeアダプタ](/ja/guides/integrations-guide/node/)がインストールされている必要があります。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:53]:
```suggestion
プロジェクトに`.dockerignore`ファイルを追加するのがベストプラクティスです。このファイルには、Dockerの`COPY`コマンドや`ADD`コマンドで無視するファイルやフォルダを記述します。これはビルドプロセスを高速化し、最終的なイメージのサイズを小さくします。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:61]:
```suggestion
このファイルは`Dockerfile`自体と同じディレクトリに置く必要があります。[詳しくは`.dockerignore`ドキュメントを読んでください](https://docs.docker.com/engine/reference/builder/#dockerignore-file)
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:67]:
```suggestion
以下のDockerfileは、サイトをビルドし、それをApache htppdによりデフォルトの設定でポート`80`にて配信します。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:101]:
```suggestion
上記のDockerfileをビルドするには、NGINX用の設定ファイルも作成する必要があります。プロジェクトのルートディレクトリに`nginx`というフォルダを作成し、その中に`nginx.conf`というファイルを作成します。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:139]:
```suggestion
以下は、Dockerの[マルチステージビルド](https://docs.docker.com/build/building/multi-stage/)を用い、ソースコードのみが変更された場合にnpmの依存関係を再インストールしないようにすることで、サイトのビルドプロセスを最適化する、より高度なDockerfileの例です。これにより、依存関係のサイズにもよりますが、ビルド時間を場合によっては数分程度短縮できます。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:146]:
```suggestion
# package.jsonとpackage-lock.jsonのみをここにコピーすることで、以下の`-deps`ステップがソースコードから独立していることが保証されます。
# したがって、ソースコードが変更されただけであれば、 `-deps`ステップはスキップされます。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:171]:
```suggestion
1. プロジェクトのルートディレクトリで以下のコマンドを実行してコンテナをビルドします。`<your-astro-image-name>`には任意の名前を使用してください。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:179]:
```suggestion
2. イメージをローカルコンテナとして実行するには、以下のコマンドを使用します。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:181]:
```suggestion
`<local-port>`を自分のマシンの使用されていないポートに置き換えてください。また `<container-port>`をDockerコンテナが公開しているポートに置き換えてください（上記の例では`4321`、`80`、`8080`）。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:187]:
```suggestion
`http://localhost:<local-port>`からサイトにアクセスできるはずです。
```

File: src/content/docs/ja/recipes/docker.mdx
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:189]:
```suggestion
3. これでウェブサイトが正常にビルドされ、コンテナにパッケージ化されたので、クラウドプロバイダにデプロイすることができます。例として[Google Cloud](/ja/guides/deploy/google-cloud/#cloud-run-ssrと静的配信)のデプロイガイドや、Dockerドキュメントの[Deploy your app](https://docs.docker.com/language/nodejs/deploy/)ページを参照してください。
```

**Review by morinokami — CHANGES_REQUESTED** [review]
@sarah11918 Added some suggestions! I think we can merge this PR once those suggestions are merged.

@piro-hiroki Please let me know if you have anything to add!

**Review by sarah11918 — APPROVED** [review]
Thank you both @piro-hiroki and @morinokami ! I am so thankful for your contributions!

**Pull request merged**
